### PR TITLE
idea: Centralized on-failure reporting for failed systemd units

### PR DESCRIPTION
**Pull request by peterhoeg**
one more

### CODE REVIEW

**Comment by peterhoeg** [comment]
@Janik-Haag

File: ideas/2024.md [hunk 1]
```diff
@@ -23,30 +23,57 @@ medium
...
 
+Prior efforts:
+We have both working crystal build infrastructure as well as a working [crystal2nix](https://github.com/nix-community/crystal2nix), so you will not be starting from scratch.
+
+### Centralized on-failure reporting for failed systemd units
+
+A systemd service supports triggering another unit if it fails through the `OnFailure` option. In NixOS we have centralized configuration and yet no way to easily opt in to common/shared `OnFailure` handlers. 
+
+With such infrastructure in place, it becomes trivial to have hosts react properly to failure and outages in addition to whatever other monitoring might be in place. 
+
```
Comment (Janik-Haag) [inline ideas/2024.md:59]:
```suggestion
A systemd service supports triggering another unit if it fails through the `OnFailure` option. In NixOS we have centralized configuration and yet no way to easily opt in to common/shared `OnFailure` handlers.

With such infrastructure in place, it becomes trivial to have hosts react properly to failure and outages in addition to whatever other monitoring might be in place.

```
just removing trailing whitspaces

File: ideas/2024.md [hunk 2]
```diff
@@ -23,30 +23,57 @@ medium
...
 Possible Mentors:
 - [@peterhoeg](https://github.com/peterhoeg)
 
 Difficulty:
 easy
 
+Prior efforts:
+We have both working crystal build infrastructure as well as a working [crystal2nix](https://github.com/nix-community/crystal2nix), so you will not be starting from scratch.
+
+### Centralized on-failure reporting for failed systemd units
```
Comment (Janik-Haag) [inline ideas/2024.md:54]: everything below line 54 should probably be it's own commit for the systemd unit reporting idea.

Comment (peterhoeg) [reply ideas/2024.md:54]: I can split it up, but the commit history here doesn't really matter, does it?
